### PR TITLE
rust: update to 1.76.0

### PR DESCRIPTION
**Pull request by krant**
Maintainer: @lu-zero 
Run tested: ARMv7, Linksys WRT3200ACM - build and run ripgrep package

Description:
- Use .xz for source archive
- Refresh patches

### CODE REVIEW

**Review by lu-zero — APPROVED** [review]
Looks fine, thank you

File: lang/rust/patches/0002-rustc-bootstrap-cache.patch [hunk 1]
```diff
@@ -38,15 +41,3 @@
          let cache_dir = cache_dst.join(key);
          if !cache_dir.exists() {
              t!(fs::create_dir_all(&cache_dir));
-@@ -671,7 +680,10 @@ download-rustc = false
-         let llvm_assertions = self.llvm_assertions;
- 
-         let cache_prefix = format!("llvm-{llvm_sha}-{llvm_assertions}");
--        let cache_dst = self.out.join("cache");
-+        let cache_dst = match env::var_os("OPENWRT_RUSTC_BOOTSTRAP_CACHE") {
```
Comment (1715173329) [inline lang/rust/patches/0002-rustc-bootstrap-cache.patch:46]: I wonder why this part was removed.

Comment (krant) [reply lang/rust/patches/0002-rustc-bootstrap-cache.patch:46]: Ouch, it's a bug I did when was manually rebasing the patches. Shame on me!
Should I prepare the fix?

Comment (1715173329) [reply lang/rust/patches/0002-rustc-bootstrap-cache.patch:46]: Would be great if you can create a PR for it ;)

Comment (krant) [reply lang/rust/patches/0002-rustc-bootstrap-cache.patch:46]: @1715173329 sorry for the hussle, PR is ready - #23581

Comment (1715173329) [reply lang/rust/patches/0002-rustc-bootstrap-cache.patch:46]: No worries! Thank you!

**Pull request merged**